File: mario_less.c
#include <cs50.h>
#include <stdio.h>

int get_height(string prompt);

// prompt user for the number between 1 and 8
int get_height(string prompt)
{
    int n;
    do
    {
        n = get_int("%s", prompt);
    }
    while (n > 8 || n < 1);
    return n;
}

int main(void)
{
    int i = get_height("Height: ");
    
    //build tha wall!
    for (int j = 0; j < i; j++)
    {
        int l, k;
        for (l = i - 2; l >= j; l--)
        {
            printf(" ");
        }
        for (k = 0; k <= j; k++)
        {
            printf("#");
        }
        printf("\n");
    }
}

File: mario_less.py
# mario_less
from cs50 import get_int


def main():
    h = get_height("Height: ")
    for i in range(h):
        for j in range(h - (i + 1)):
            print(" ", end="")
        for k in range(i + 1):
            print("#", end="")
        print()


def get_height(prompt):
    # get height between 1 and 8
    while True:
        n = get_int(prompt)
        if n > 0 and n < 9:
            break
    return n


if __name__ == "__main__":
    main()

